pytorch_lightning/trainer/trainer.py

```diff
@@ -601,7 +598,29 @@ def get_dataloaders(self, model):
 
             If you want each process to load the full dataset, ignore this warning.
             """
-            warnings.warn(msg)
+            if msg not in self.shown_warnings and self.proc_rank == 0:
+                self.shown_warnings.add(msg)
+                warnings.warn(msg)
+
+    def init_val_dataloader(self, model):
+        """
+        Dataloaders are provided by the model
+        :param model:
+        :return:
+        """
+        self.val_dataloader = model.val_dataloader
+
+        # enable multiple validation loaders
```

Comment:
all this looks very similar for both test/valid... consider have a general func:
```
@staticmethod
def _nb_batches(dataloader, percent_check):
     have_loaders = dataloader is not None
     if have_loaders and not isinstance(dataloader, list):
         dataloader = [self.dataloader]

    # determine number of validation batches, datasets could be none, 1 or 2+
    if dataloader is not None:
        nb_batches = sum(len(dl) for dl in dataloader)
        nb_batches = int(nb_batches * percent_check)
        nb_batches = max(1, nb_batches)
    else:
        nb_batches = 0
    return dataloader, nb_batches
```
and later just:
```
self.val_dataloader, self.nb_val_batches = self._nb_batches(self.val_dataloader, self.val_percent_check)
```
and alternatively for `test` so it is clear thet the process is the same...